lib/makara/config_parser.rb

```diff
@@ -26,6 +26,114 @@ class ConfigParser
       :sticky => true
     }
 
+    # ConnectionUrlResolver is borrowed from Rails 4-2 since its location and implementation
+    # vary slightly among Rails versions, but the behavior is the same.  Thus, borrowing the
+    # class should be the most future-safe way to parse a database url.
+    #
+    # ActiveRecord::Base::ConnectionSpecification::Resolver.send(:connection_url_to_hash, url_config[:url])
+    # https://github.com/rails/rails/blob/3-2-stable/activerecord/lib/active_record/connection_adapters/abstract/connection_specification.rb#L60-L77
+    # ActiveRecord::ConnectionAdapters::ConnectionSpecification::Resolver.send(:connection_url_to_hash, url_config[:url])
+    # https://github.com/rails/rails/blob/4-0-stable/activerecord/lib/active_record/connection_adapters/connection_specification.rb#L68-L92
+    # ActiveRecord::ConnectionAdapters::ConnectionSpecification::ConnectionUrlResolver.new(url).to_hash
+    # https://github.com/rails/rails/blob/4-1-stable/activerecord/lib/active_record/connection_adapters/connection_specification.rb#L17-L121
+    # ActiveRecord::ConnectionHandling::MergeAndResolveDefaultUrlConfig.new(url_config).resolve
+    # ActiveRecord::ConnectionAdapters::ConnectionSpecification::ConnectionUrlResolver.new(url).to_hash
+    # https://github.com/rails/rails/blob/4-2-stable/activerecord/lib/active_record/connection_handling.rb#L60-L81
+    # https://github.com/rails/rails/blob/97b980b4e61aea3cee429bdee4b2eae2329905cd/activerecord/lib/active_record/connection_handling.rb#L60-L81
+    #
+    # Expands a connection string into a hash.
+    class ConnectionUrlResolver # :nodoc:
```

Comment:
I'm open to changing this to however you like, new file, less comments, less encapsulation,  etc